lua/precognition/init.lua

```diff
@@ -3,41 +3,73 @@ local vm = require("precognition.vertical_motions")
 
 local M = {}
 
----@alias SupportedHints "'^'" | "'b'" | "'w'" | "'$'"
----@alias SupportedGutterHints "G" | "gg" | "{" | "}"
+---@alias Dollar "$"
+---@alias PrevParagraph "{"
+---@alias NextParagraph "}"
+
+---@class HintOpts
```

Comment:
I'd recommend namespacing these behind `Precognition.*` to avoid conflicting with types from other plugins.

Reply:
I don't think i actually need them, i have removed them entirely